GitHub.Collectors/Authentication/GitHubAppAuthentication.cs

```diff
@@ -64,42 +59,54 @@ public Task<string> GetAuthorizationHeaderAsync()
+        protected override async Task<string> FindInstallationId()
         {
             if (OrgNameToInstallationIdMap.TryGetValue(this.organization, out string cachedInstallationId))
             {
                 return cachedInstallationId;
             }
             else
             {
-                string requestUri = $"https://{this.apiDomain}/app/installations?per_page=100";
-                BatchingGitHubHttpRequest batchingHttpClient = new BatchingGitHubHttpRequest(httpClient, requestUri, "App.Installations", new List<HttpResponseSignature>());
-
-                IAuthentication jwtAuthentication = new JwtAuthentication(this.Identity + "-jwt", jwt);
+                JArray installations = await GetAppInstallations().ConfigureAwait(false);          
                 string targetInstallationId = null;
 
-                while (batchingHttpClient.HasNext)
+                foreach (JObject installation in installations)
                 {
-                    HttpResponseMessage response = await batchingHttpClient.NextResponseAsync(jwtAuthentication).ConfigureAwait(false);
+                    string orgName = installation.SelectToken("$.account.login").Value<string>();
+                    string installationId = installation.SelectToken("$.id").Value<string>();
+                    OrgNameToInstallationIdMap[orgName] = installationId;
 
-                    JArray responseBody = await HttpUtility.ParseAsJArrayAsync(response).ConfigureAwait(false);
-                    foreach (JObject responseItem in responseBody)
+                    if (orgName.Equals(this.organization))
                     {
-                        string orgName = responseItem.SelectToken("$.account.login").Value<string>();
-                        string installationId = responseItem.SelectToken("$.id").Value<string>();
-                        OrgNameToInstallationIdMap[orgName] = installationId;
-
-                        if (orgName.Equals(this.organization))
-                        {
-                            targetInstallationId = installationId;
-                        }
+                        targetInstallationId = installationId;
                     }
                 }
 
                 return targetInstallationId ?? throw new FatalException($"Could not find installation id for organization {organization} with AppId {this.appId}");
             }
         }
 
+        public async Task<JArray> GetAppInstallations()
+        {
+            string jwt = CreateJwt();
+            string requestUri = $"https://{this.apiDomain}/app/installations?per_page=100";
+            BatchingGitHubHttpRequest batchingHttpClient = new BatchingGitHubHttpRequest(this.httpClient, requestUri, "App.Installations", new List<HttpResponseSignature>());
+            IAuthentication jwtAuthentication = new JwtAuthentication(this.Identity + "-jwt", jwt);
+            JArray installations = new JArray();
```

Comment:
>JArray installations = new JArray(); [](http://example.com/codeflow?start=12&length=36)

Any reason to prefer JArray over List<JObject> (as you have below commented out)? Just wondering.

Reply:
Not sure why I did this. I have changed back to List